File: Data Structures & Algorithms/subarray-product-less-than-k/submission-0.java
class Solution {
    public int numSubarrayProductLessThanK(int[] nums, int k) {

        int l=0, res=0;
        long prod = 1;
        for(int r=0; r < nums.length; r++){
            prod *= nums[r];
            while(l <=r && prod >=k){
                prod /= nums[l];
                l++;
            }

            res += r - l +1;
        }

        return res;
        
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        indices = {}

        for i,n in enumerate(nums):
            indices[n] = i

        for i,n in enumerate(nums):
            diff = target - n
            if diff in indices and indices[diff] != i: #same index
                return [i,indices[diff]]        